docs/notebooks/basic-usage.ipynb

```diff
@@ -69,23 +69,31 @@
    "cell_type": "code",
    "execution_count": null,
    "id": "846bfb43",
-   "metadata": {},
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
```

Comment:
Omg 🤯

Reply:
And also, can you change the sys.path in the first cell to point to `src`. I.e., it should read

```
sys.path.insert(0, str(pathlib.Path().absolute().parents[1] / "src"))
```

or


```
sys.path.insert(0, str(pathlib.Path().absolute().parent.parent / "src"))
```

Reply:
(that’s the hidden cell)

Reply:
@christophfink : Actually, do we need / want to have the first cell in the Notebook? I understand the purpose of it, but it can actually cause issues when users test the tool in Binder environment. I think the idea with the `basic_usage` Notebook is to reflect the package which is released and stable? If we use the sys.path approach and user executes the hidden cell (which is very likely) in the Binder, then the codes will reflect the situation of `main` branch (not released version). This can cause conflicts and is a bit risky. Hence, should we keep this Notebook isolated from the development, i.e. remove the hidden cell entirely? We could perhaps have a separate dev notebook for testing which is separate from the basic tutorial.

Reply:
Hm, that’s true, but then we have to install the package (right now it does not find the `r5py` module)

Reply:
And also on rtd

Reply:
(I’m pretty sure that sphinx’ conf.py is not considered inside the notebook)

Reply:
At the moment, `r5py` is actually already installed for the Binder environment from conda-forge: https://github.com/r5py/r5py/blob/main/binder/environment.yml

Reply:
We could do the same for RTD as well, it shouldn't be a problem.

Reply:
or, even worse, for the last version that is on conda (which is very likely to not be the one that triggered rtd)

Reply:
Hmm, right so you want the RTD latest to reflect the dev-version (which makes sense)? And the stable shows the released version. In this case, I guess we want the Binder to reflect the Notebook from the release, not the one which is currently up-to-date with main branch. I guess this would make most sense. Now just need to figure out how to configure this 😄

Reply:
> Hmm, right so you want the RTD latest to reflect the dev-version (which makes sense)? And the stable shows the released version. In this case, I guess we want the Binder to reflect the Notebook from the release, not the one which is currently up-to-date with main branch? I guess this would make most sense. Now just need to figure out how to configure this smile

Okay, just checked this and it should be easy to configure the Binder to use the Notebook from a released/tagged version. 👍🏻 Still need to test whether we can also configure this similarly in conf.py.

Reply:
Okay, seems to work as well configuring this similarly in conf.py. Hence, the workflow would be such that for each **release** (which goes in RTD under the `stable` version) we should do following:

1. Update the Binder launch button in repo Readme.md to point to the **tagged/released** version (e.g. `v0.3.3`).
2. Update the `"repository_branch"` in `html_theme_options` in `conf.py` to point to the tagged/released version (e.g. `v0.3.3`).

After these steps, those should be replaced with the original ones to get the `latest` version in RTD to work correctly with the latest r5py codebase from `main`.

Reply:
I think this update cycle could be automated in [release-and-deploy-v-tags.yml](https://github.com/r5py/r5py/blob/main/.github/workflows/release-and-deploy-v-tags.yml) because otherwise it is very likely that we screw things up with these when we release a new version of the package?

Reply:
How does this sound @christophfink ?

Reply:
I’m thinking that the version should always be the correct one without any added configuration, if we simply have the notebooks load `r5py` from the `src` directory. No need to automate anything, then - or am I thinking this wrongly? RTD is always referencing the respective stable or latest code. So simply changing the `sys.path` to keep pointing to the source directory works for RTD and locally cloned repos.

When it comes to binder, we should maybe take it into a new issue, because this here is about jupytext. This is getting a bit too complex and off-topic for my taste in this PR.

Reply:
Could we here just correct the sys.path, and then talk about how to keep binder respecting the stable/latest branch in a new issue?

Reply:
(which might overthrow the sys.path mechanism, but it will be documented better why and how)

Reply:
Yes, I was here now discussing more about the logic for Binder (which came as a consequence of reorganizing the docs). But as you say, it is indeed a separate topic from this PR and I opened a new issue for this in #122. RTD will always show the correct document.

Reply:
> Could we here just correct the sys.path, and then talk about how to keep binder respecting the stable/latest branch in a new issue?

This is started in #122